FaunaDB.Client/Client/DefaultClientIO.cs

```diff
@@ -34,11 +36,14 @@ public DefaultClientIO(Uri domain, TimeSpan timeout, string secret)
             client.DefaultRequestHeaders.AcceptEncoding.Add(new StringWithQualityHeaderValue("gzip"));
             client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
 
-            authHeader = AuthString(secret);
+            authHeader = AuthHeader(secret);
         }
 
         public IClientIO NewSessionClient(string secret) =>
-            new DefaultClientIO(client, secret);
+            new DefaultClientIO(client, AuthHeader(secret), useRecursiveRefs);
+
+        public IClientIO NewWithRecursiveRefs(bool useRecursive) =>
```

Comment:
I understand that is unlike people will bounce back and forth, but there must be a way to upgrade the version. All tests so far is based on 2.0 and it needs to keep working. What's your suggestion? Add the api version on constructor? And what about `NewSessionClient` method? Add the version there as well?